Gems/Atom/Asset/ImageProcessingAtom/Code/Source/Thumbnail/ImageThumbnail.cpp

```diff
@@ -85,11 +89,17 @@ namespace ImageProcessingAtom
         {
             if (m_state == State::Ready && m_assetIds.find(assetId) != m_assetIds.end())
             {
-                m_state = State::Unloaded;
-                Load();
+                AZ::Data::AssetBus::MultiHandler::BusConnect(assetId);
             }
         }
 
+        void ImageThumbnail::OnAssetReloaded(AZ::Data::Asset<AZ::Data::AssetData> asset)
+        {
+            AZ::Data::AssetBus::MultiHandler::BusDisconnect(asset.GetId());
+            m_state = State::Unloaded;
+            Load();
+        }
```

Comment:
Just verifying, if a second reload for the same asset gets triggered, the "BusDisconnect" here doesn't cause a problem? It feels like there might be a potential race condition if things would trigger like this:
* OnCatalogAssetChanged
* OnCatalogAssetChanged
* OnAssetReloaded
* OnAssetReloaded